### PR TITLE
sum avg

### PR DESCRIPTION
**Pull request by MarkDuckworth**
Adds support for sum and average aggregate queries. See [go/firestore-sum-avg-online-api](http://goto.google.com/firestore-sum-avg-online-api) (internal only).

### CODE REVIEW

**Review by markarndt — APPROVED** [review]

**Review by hsubox76 — COMMENTED** [review]
Can you add a description in the PR? You can link to the API proposal or design doc and just say "(internal link)" after it.

**Review by hsubox76 — COMMENTED** [review]
Also looks like there's some test failures directly related to this change.

File: .changeset/beige-oranges-eat.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@firebase/firestore": minor
```
[the remark this answers is not in the document]
Comment (MarkDuckworth) [reply .changeset/beige-oranges-eat.md:2]: Added, thank you.

**Review by hsubox76 — APPROVED** [review]

File: packages/firestore/src/lite-api/aggregate_types.ts [hunk 2]
```diff
@@ -32,25 +33,30 @@ export class AggregateField<T> {
...
    * @param _internalFieldPath Optionally specifies the field that is aggregated.
    * @internal
    */
   constructor(
-    // TODO (sum/avg) make aggregateType public when the feature is supported
     readonly _aggregateType: AggregateType = 'count',
     readonly _internalFieldPath?: InternalFieldPath
-  ) {}
+  ) {
+    this.aggregateType = _aggregateType;
```
Comment (ehsannas) [inline packages/firestore/src/lite-api/aggregate_types.ts:49]: the class now has both `this._aggregateType` and `this.aggregateType`.

File: common/api-review/firestore-lite.api.md [hunk 3]
```diff
@@ -19,11 +19,15 @@ export type AddPrefixToKeys<Prefix extends string, T extends Record<string, unkn
 
 // @public
 export class AggregateField<T> {
+    readonly aggregateType: AggregateType;
```
Comment (ehsannas) [inline common/api-review/firestore-lite.api.md:22]: the `field` public member is missing

File: packages/firestore/src/lite-api/aggregate_types.ts (hunk 2 above)
Comment (MarkDuckworth) [reply packages/firestore/src/lite-api/aggregate_types.ts]: fixed.

File: common/api-review/firestore-lite.api.md (hunk 3 above)
Comment (MarkDuckworth) [reply common/api-review/firestore-lite.api.md:22]: Resolved in out of band conversation. Not including the `field` public memeber.

**Pull request merged**